packages/ui-components/src/components/form.tsx

```diff
@@ -677,6 +691,144 @@ export function FormComponent(props: IFormComponentProps): JSX.Element {
   };
 
   return (
-    <Form templates={templates} formContext={formContext as any} {...others} />
+    <Form
+      templates={templates as any}
+      formContext={formContext as any}
+      widgets={{ ...customWidgets, ...widgets }}
+      {...others}
+    />
+  );
+}
+
+function CustomCheckbox<
+  T = any,
+  S extends StrictRJSFSchema = RJSFSchema,
+  F extends FormContextType = any
+>({
+  schema,
+  id,
+  value,
+  disabled,
+  readonly,
+  label = '',
+  onChange,
+  onBlur,
+  onFocus
+}: WidgetProps<T, S, F>) {
+  // Because an unchecked checkbox will cause html5 validation to fail, only add
+  // the "required" attribute if the field value must be "true", due to the
+  // "const" or "enum" keywords
+  const required = schemaRequiresTrueValue<S>(schema);
+  const _onChange = ({
+    target: { checked }
+  }: React.ChangeEvent<HTMLElement & { checked: boolean }>) =>
+    onChange(checked);
+  const _onBlur = ({
+    target
+  }: React.FocusEvent<HTMLElement & { value: string }>) =>
+    onBlur(id, target && target.value);
+  const _onFocus = ({
+    target
+  }: React.FocusEvent<HTMLElement & { value: string }>) =>
+    onFocus(id, target && target.value);
+  return (
+    <Checkbox
+      id={id}
+      name={id}
+      checked={typeof value === 'undefined' ? false : Boolean(value)}
+      required={required}
+      disabled={disabled || readonly}
+      onChange={_onChange}
+      onBlur={_onBlur}
+      onFocus={_onFocus}
+      aria-describedby={ariaDescribedByIds<T>(id)}
+    >
+      {label}
+    </Checkbox>
+  );
+}
+
+export const customWidgets = {
+  CheckboxWidget: CustomCheckbox,
+  SelectWidget: CustomSelect,
+};
+
+function CustomSelect<
+  T = any,
+  S extends StrictRJSFSchema = RJSFSchema,
+  F extends FormContextType = any
+  >({
+  id,
+  options,
+  label,
+  hideLabel,
+  value,
+  required,
+  disabled,
+  readonly,
+  multiple = false,
+  autofocus = false,
+  rawErrors = [],
+  onChange,
+  onBlur,
+  onFocus,
+  schema,
+  placeholder,
+}: WidgetProps<T, S, F>) {
+  const { enumOptions, enumDisabled, emptyValue: optEmptyVal } = options;
+
+  const selectedIndexes = enumOptionsIndexForValue<S>(value, enumOptions, multiple);
+  let selectedIndexesAsArray: string[] = [];
+
+  if (typeof selectedIndexes === 'string') {
+    selectedIndexesAsArray = [selectedIndexes];
+  } else if (Array.isArray(selectedIndexes)) {
+    selectedIndexesAsArray = selectedIndexes.map((index) => String(index));
+  }
+
+  const dropdownValue = selectedIndexesAsArray
+    .map((index) => (enumOptions ? enumOptions[Number(index)].label : undefined))
+    .join(', ');
+
+  const _onBlur = () => onBlur(id, selectedIndexes);
+  const _onFocus = () => onFocus(id, selectedIndexes);
+  const _onChange = (_: any, data: OptionOnSelectData) => {
+    const newValue = getValue(data, multiple);
+    return onChange(enumOptionsValueForIndex<S>(newValue, enumOptions, optEmptyVal));
+  };
```

Comment:
The callback `OnOptionSelect` does not exist on the toolkit.

You gonna have to use `onChange(event) {}`. And to determine the `newValue`, you should be able to do something like

```ts
const _onChange = (event) => {
  const newValue = multiple ? (event.target as any).selectedOptions : (event.target as any).value;
}
```